pym/bob/generators/VisualStudioCode.py

```diff
@@ -0,0 +1,175 @@
+        self.resources = getCorrectPathList(recipesRoot, scan.resources)
+        self.incPaths =  getCorrectPathList(recipesRoot, scan.incPaths)
+        self.dependencies = scan.dependencies
+        self.runTargets = getCorrectPathList(recipesRoot, scan.runTargets)
+        includeList += self.incPaths
+
+class VsCodeGenerator(CommonIDEGenerator):
+    def __init__(self):
+        super().__init__("vsCode", "Generate Visual Studio Code workspace")
+        self.parser.add_argument('--sort', action='store_true', help="Sort the dependencies by name (default: unsorted)")
+
+    def configure(self, package, argv):
+        super().configure(package, argv)
+
+    def generate(self, extra, bobRoot):
+        super().generate()
+
+        # gather root paths
+        bobPwd = Path(os.getcwd())
+        if isMsys():
+            if os.getenv('WD') is None:
+                raise BuildError("Cannot create Visual Studio Code project for Windows! MSYS2 must be started by msys2_shell.cmd script!")
+            msysRoot = PureWindowsPath(os.getenv('WD')) / '..' / '..'
+            winPwd = PureWindowsPath(os.popen('pwd -W').read().strip())
+            shell = str(msysRoot / "msys2_shell.cmd")
+            shellArgs = (MSYS_ARGUMENTS + [str(winPwd), "-l", "-c"])
+        else:
+            winPwd = bobPwd
+            winDestination = Path(self.destination).resolve()
+            shell = "cmd.exe"
+            shellArgs = []
+
+        JSON_WORKSPACE_TEMPLATE["tasks"]["options"]["cwd"] = str(winPwd)
+        JSON_WORKSPACE_TEMPLATE["tasks"]["options"]["shell"]["executable"] = shell
+        JSON_WORKSPACE_TEMPLATE["tasks"]["options"]["shell"]["args"] = shellArgs
+
+        includeList = self.prependIncludeDirectories
+        projects = {
+            name : Project(winPwd, scan, includeList)
+            for name,scan in self.packages.items()
+        }
+
+        JSON_WORKSPACE_TEMPLATE["settings"]["C_Cpp.default.includePath"] = includeList + self.appendIncludeDirectories
+
+        workspaceProjectList = []
+        for name,project in sorted(projects.items()) if self.args.sort else projects.items():
+            workspaceProjectList.append({
+                "name": name,
+                "path": str(project.workspacePath)
+            })
+
+        JSON_WORKSPACE_TEMPLATE["folders"] = workspaceProjectList
+        JSON_WORKSPACE_TEMPLATE["tasks"]["tasks"] = self.__getTaskList(bobRoot, "/".join(self.rootPackage.getStack()))
+
+        self.updateFile(os.path.join(self.destination, self.projectName+".code-workspace"),
+                json.dumps(JSON_WORKSPACE_TEMPLATE, indent=4),
+                encoding="utf-8", newline='\r\n')
+
+    def __getTaskList(self, bobRoot, package):
+        taskList = []
+        for entry in TASK_LIST:
+            task = JSON_TASK_TEMPLATE.copy()
+            task["label"] = entry["label"]
+            task["command"]= " ".join([bobRoot, "dev", package] + entry["args"])
```

Comment:
For the MSYS environment the code with the single command line without arguments is the only version which is working. I think this is because the shell is called with a lot of arguments.
But I made case for msys and the other environment right now.

Reply:
Tested the latest version with WSL2 environment and the tasks are working.